arm-compute/2016-09-30/swagger/containerService.json

```diff
@@ -0,0 +1,598 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerServiceClient",
+    "description": "The Container Service Client.",
+    "version": "2016-09-30"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerService/containerServices": {
+      "get": {
+        "tags": [
+          "ContainerServices"
+        ],
+        "operationId": "ContainerServices_List",
+        "description": "Lists all container services in a subscription",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ContainerServiceListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/containerServices/{containerServiceName}": {
+      "put": {
+        "tags": [
+          "ContainerServices"
+        ],
+        "operationId": "ContainerServices_CreateOrUpdate",
```

Comment:
>ContainerServices_CreateOrUpdate [](start = 24, length = 32)

Did you guys consider separating Create and Update operations by implementing PATCH on the server side? The issue with this coupling is that for Update operation caller will have to provide original objects and all required field which are 
"masterProfile",
"agentPoolProfiles",
"linuxProfile"

Whereas in the patch you can have only updatable fields combined in a different model. And all of them should be optional to avoid inconvenience for the end users.